omod/src/test/java/org/openmrs/module/fhir2/providers/r3/ListFhirResourceProviderWebTest.java

```diff
@@ -76,4 +92,91 @@ public void shouldReturn404IfListNotFound() throws Exception {
+		
+		MockHttpServletResponse response = post("/List").jsonContent(listJson).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isCreated());
+		assertThat(response.getStatus(), is(201));
+	}
+	
+	@Test
+	public void updateList_shouldUpdateRequestedList() throws Exception {
+		org.hl7.fhir.r4.model.ListResource listResource = new org.hl7.fhir.r4.model.ListResource();
+		listResource.setId(LIST_UUID);
+		String listJson;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(JSON_UPDATE_LIST_PATH)) {
+			Objects.requireNonNull(is);
+			listJson = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+		
+		when(cohortFhirListService.update(any(String.class), any(org.hl7.fhir.r4.model.ListResource.class)))
+		        .thenReturn(listResource);
+		
+		MockHttpServletResponse response = put("/List/" + LIST_UUID).jsonContent(listJson).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+	}
+	
+	@Test
+	public void updateList_shouldErrorForNoId() throws Exception {
+		String listJson;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(JSON_UPDATE_LIST_NO_ID_PATH)) {
+			Objects.requireNonNull(is);
+			listJson = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+		
+		MockHttpServletResponse response = put("/List/" + LIST_UUID).jsonContent(listJson).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isBadRequest());
+		assertThat(response.getContentAsString(), containsStringIgnoringCase("body must contain an ID element for update"));
+	}
+	
+	@Test
+	public void updateList_shouldErrorForIdMissMatch() throws Exception {
+		String listJson;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(JSON_UPDATE_LIST_WRONG_ID_PATH)) {
+			Objects.requireNonNull(is);
+			listJson = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+		
+		MockHttpServletResponse response = put("/List/" + UNKNOWN_UUID).jsonContent(listJson).accept(FhirMediaTypes.JSON)
+		        .go();
+		
+		assertThat(response, isBadRequest());
+		assertThat(response.getContentAsString(),
+		    containsStringIgnoringCase("body must contain an ID element which matches the request URL"));
+	}
+	
+	@Test
+	public void deleteList_shouldDeleteList() throws Exception {
+		org.hl7.fhir.r4.model.ListResource listResource = new org.hl7.fhir.r4.model.ListResource();
+		listResource.setId(LIST_UUID);
+		when(cohortFhirListService.delete(LIST_UUID)).thenReturn(listResource);
+		
+		MockHttpServletResponse response = delete("/List/" + LIST_UUID).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), equalTo(FhirMediaTypes.JSON.toString()));
+	}
+	
+	@Test
+	public void deleteList_shouldReturn404ForNonExistingList() throws Exception {
+		when(cohortFhirListService.get(UNKNOWN_UUID)).thenReturn(null);
+		
+		MockHttpServletResponse response = get("/List/" + UNKNOWN_UUID).accept(FhirMediaTypes.JSON).go();
```

Comment:
Don't you think you should be using delete instead of get in this test ?

Reply:
exactly